### ISSUE TITLE
Is dictionary for "steps" allowed?

### ISSUE THREAD

**Issue by otiai10** [common-workflow-language/common-workflow-language#513]
As I faced [this example](https://github.com/common-workflow-language/common-workflow-language/blob/master/v1.0/v1.0/count-lines2-wf.cwl#L16-L50), "steps" is described as dict of yaml (with key names `step1` and `step2`).
On the other hand, checking [this document](http://www.commonwl.org/v1.0/Workflow.html#Workflow), "steps" should be array (`array<WorkflowStep>`).

Which is correct? or am I wrong?

**Comment by tetron** [comment]
You're right, there is a mistake in the documentation for the specification.

Here's the section from the formal schema: https://github.com/common-workflow-language/common-workflow-language/blob/master/v1.0/Workflow.yml

```
    - name: steps
      doc: |
        The individual steps that make up the workflow.  Each step is executed when all of its
        input data links are fufilled.  An implementation may choose to execute
        the steps in a different order than listed and/or execute steps
        concurrently, provided that dependencies between steps are met.
      type:
        - type: array
          items: "#WorkflowStep"
      jsonldPredicate:
          mapSubject: id
```

The `mapSubject` specifies an identifier map: http://www.commonwl.org/v1.0/SchemaSalad.html#Identifier_maps

This means a dictionary allowed as input, but is normalized to an array for validation and downstream processing.

The documentation HTML is generated using this code: https://github.com/common-workflow-language/schema_salad/blob/master/schema_salad/makedoc.py

Most likely there is a bug in the documentation generator.

Thank you for catching this error.

**Comment by otiai10** [comment]
@tetron thanks.
